src/plexil-adapter/OwInterface.h

```diff
@@ -115,6 +159,11 @@ class OwInterface
   ros::Subscriber* m_rulSubscriber;
   ros::Subscriber* m_guardedMoveSubscriber;
 
+  ros::Subscriber* m_systemFaultMessagesSubscriber;
+  ros::Subscriber* m_armFaultMessagesSubscriber;
+  ros::Subscriber* m_powerFaultMessagesSubscriber;
+  ros::Subscriber* m_ptFaultMessagesSubscriber;
```

Comment:
Use [`std::unique_ptr`](https://en.cppreference.com/w/cpp/memory/unique_ptr) instead of raw pointers
```suggestion
  std::unique_ptr<ros::Subscriber> m_systemFaultMessagesSubscriber;
  std::unique_ptr<ros::Subscriber> m_armFaultMessagesSubscriber;
  std::unique_ptr<ros::Subscriber> m_powerFaultMessagesSubscriber;
  std::unique_ptr<ros::Subscriber> m_ptFaultMessagesSubscriber;
```

Reply:
just tried and subscriber won't compile if we use unique ptr

Reply:
You may need to read a little on `std::unique_ptr` but in short you should the initialization code to the following:
```bash
m_<...>Subscriber.reset( new ros::Subscriber( ... ) );
```